<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>editRuleLibrary</class>
 <widget class="QDialog" name="editRuleLibrary">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>925</width>
    <height>575</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QFrame" name="gridFrame">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QPushButton" name="addRulesButton">
          <property name="text">
           <string>add</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="QScrollArea" name="rulesScrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>350</width>
             <height>477</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QTreeWidget" name="treeWidget"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>538</width>
          <height>514</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QFrame" name="ruleSettingsFrame">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QFrame" name="gridFrame">
              <layout class="QGridLayout" name="gridLayout">
               <item row="2" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>comments</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <spacer name="horizontalSpacer_3">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="2" column="1" colspan="2">
                <widget class="QTextEdit" name="textEdit"/>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="lineEdit"/>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>ruleName</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="triggerGridFrame">
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="TriggerLabel">
                 <property name="font">
                  <font>
                   <pointsize>13</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Trigger</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="conditionTypeLabel">
                 <property name="text">
                  <string>condition type</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="conditionTypeComboBox">
                 <item>
                  <property name="text">
                   <string>state</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>event</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>time</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="monitiorObjectLabel">
                 <property name="text">
                  <string>monitor object</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1" colspan="2">
                <widget class="QComboBox" name="monitorObjectComboBox"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="conditionLabel">
                 <property name="text">
                  <string>condition</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="ConditionOperatorComboBox">
                 <item>
                  <property name="text">
                   <string>&gt;</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>&lt;</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>=</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>≥</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>≤</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>≠</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QLineEdit" name="conditionValueLineEdit"/>
               </item>
               <item row="3" column="3">
                <widget class="QLabel" name="conditionUnitLabel">
                 <property name="text">
                  <string>token(s)</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="durationLabel">
                 <property name="text">
                  <string>duration</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QLineEdit" name="durationValueLineEdit"/>
               </item>
               <item row="4" column="2">
                <widget class="QComboBox" name="durationUnitComboBox">
                 <item>
                  <property name="text">
                   <string>ms</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>s</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>min</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>h</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="actionGridFrame">
              <layout class="QGridLayout" name="actionGridLayout" columnstretch="0,0,0">
               <item row="1" column="0">
                <widget class="QLabel" name="actionTypeLabel">
                 <property name="text">
                  <string>action type</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="ActionLabel">
                 <property name="font">
                  <font>
                   <pointsize>13</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Action</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="3" column="1" colspan="2">
                <widget class="QComboBox" name="Target2ComboBox"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="target2Label">
                 <property name="text">
                  <string>to</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1" colspan="2">
                <widget class="QComboBox" name="Target1ComboBox"/>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="actionTypeComboBox">
                 <item>
                  <property name="text">
                   <string>create</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>delete</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>update</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="target1Label">
                 <property name="text">
                  <string>target</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="connectionGridFrame">
              <layout class="QGridLayout" name="gridLayout_3">
               <item row="0" column="2">
                <widget class="QPushButton" name="pushButton">
                 <property name="text">
                  <string>Add</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="connectionLabel">
                 <property name="font">
                  <font>
                   <pointsize>13</pointsize>
                   <weight>75</weight>
                   <bold>true</bold>
                  </font>
                 </property>
                 <property name="text">
                  <string>Connections</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>editRuleLibrary</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>editRuleLibrary</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
